temporalcli/commands.operator_namespace.go

```diff
@@ -0,0 +1,335 @@
+	}
+	cctx.Printer.Println(color.GreenString("Namespace %s has been deleted.", nsName))
+	return nil
+}
+
+func (c *TemporalOperatorNamespaceDescribeCommand) run(cctx *CommandContext, args []string) error {
+	nsName, nsID, err := getNamespaceFromIDArgs(cctx, c.NamespaceId, args)
+	if err != nil {
+		return err
+	}
+
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.WorkflowService().DescribeNamespace(cctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: nsName,
+		Id:        nsID,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to describe namespace: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+	return printNamespaces(cctx, resp)
+}
+
+func (c *TemporalOperatorNamespaceListCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+
+	var nextPageToken []byte
+	for {
+		resp, err := cl.WorkflowService().ListNamespaces(cctx, &workflowservice.ListNamespacesRequest{
+			NextPageToken: nextPageToken,
+			PageSize:      100,
+		})
+		if err != nil {
+			return fmt.Errorf("failed listing namespaces: %w", err)
+		}
+
+		if cctx.JSONOutput {
+			// For JSON we are going to dump one line of JSON per execution
+			_ = cctx.Printer.PrintStructured(resp.Namespaces, printer.StructuredOptions{})
+		} else {
+			_ = printNamespaces(cctx, resp.Namespaces...)
+		}
+
+		nextPageToken = resp.GetNextPageToken()
+		if len(nextPageToken) == 0 {
+			return nil
+		}
+	}
+}
+
+func (c *TemporalOperatorNamespaceUpdateCommand) run(cctx *CommandContext, args []string) error {
+	nsName := args[0]
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	var updateRequest *workflowservice.UpdateNamespaceRequest
+
+	if c.PromoteGlobal {
+		fmt.Printf("Will promote local namespace to global namespace for:%s, other flag will be omitted. "+
+			"If it is already global namespace, this will be no-op.\n", nsName)
+		updateRequest = &workflowservice.UpdateNamespaceRequest{
+			Namespace:        nsName,
+			PromoteNamespace: true,
+		}
+	} else if len(c.ActiveCluster) > 0 {
+		fmt.Printf("Will set active cluster name to: %s, other flag will be omitted.\n", c.ActiveCluster)
+		replicationConfig := &replication.NamespaceReplicationConfig{
+			ActiveClusterName: c.ActiveCluster,
+		}
+		updateRequest = &workflowservice.UpdateNamespaceRequest{
+			Namespace:         nsName,
+			ReplicationConfig: replicationConfig,
+		}
+	} else {
+		resp, err := cl.WorkflowService().DescribeNamespace(cctx, &workflowservice.DescribeNamespaceRequest{
+			Namespace: nsName,
+		})
+		if err != nil {
+			switch err.(type) {
+			case *serviceerror.NamespaceNotFound:
+				return err
+			default:
+				return fmt.Errorf("namespace update failed: %w", err)
+			}
+		}
```

Comment:
Kinda strange Go here. This is clearer IMO:

```suggestion
		if _, ok := err.(*serviceerror.NamespaceNotFound); ok {
			return err
		} else if err != nil {
			return fmt.Errorf("namespace update failed: %w", err)
		}
```

Then again, not sure why we can't just `return fmt.Errorf("failed describing namespace for update: %w", err)` and not have the if/else. I think the old CLI had a different exit code for not found which we don't do anymore. Same for this same logic on the error later.

Reply:
Yeah right. I just removed this check.